### ISSUE TITLE
Editor suggestion for parsed map type is incorrect

### ISSUE THREAD

**Issue by yeenow123** [uw-labs/bloomrpc#20]
With the following proto file and map types

```
syntax = "proto3";

import "google/api/annotations.proto";

package com.someprofile;

service UserDataApi {
    rpc get(ReadRequest) returns (ReadResponse) {
        option (google.api.http) = {
        get: "/v1/read"
      };
    }

    rpc put(WriteRequest) returns (WriteResponse) {
        option (google.api.http) = {
        put: "/v1/write"
        body: "*"
      };
    }
}

message ReadRequest {
    repeated string userIds = 1;
    repeated string fields = 2;
}

message ReadResponse {
    map<string, InternalData> internalData = 1;
    map<string, ThirdPartyData>  thirdPartyData = 2;
}

message WriteRequest {
    map<string, InternalData> internalData = 1;
    map<string, ThirdPartyData> thirdPartyData = 2;
    int64 ttl = 3;
}

message WriteResponse {
    string userId = 1;
    bool status = 2;
}

message InternalData {
    map<string, string> data = 1;
}

message ThirdPartyData {
    map<string, Segments> data = 1;

    message Segments {
        repeated string segments = 1;
    }
}
```

The initial editor suggestion looks like the following.  It seems to be an additional nesting or something

```
{
  "internalData": {
    "data": {
      "Hello": "Hello"
    }
  },
  "thirdPartyData": {
    "data": {
      "segments": [
        "Hello"
      ]
    }
  },
  "ttl": 20
}
```

The working input actually looks like this:

```
{  
   "internalData":{  
      "user1":{  
         "data":{  
            "key1":"zak",
            "key2":"def"
         }
      }
   },
   "thirdPartyData":{
       "user1":{
           "data":{
               "key3": { "segments": [ "891", "456" ] }
           }
       }
   },
   "ttl":86400
}
```

**Comment by fenos** [comment]
Mmmh interesting, I completely miss interpreted the proto structure, thanks for the snippet that surely will make the fix more effective. 

Those maps are not easy to parse 😆 but we’ll get there eventually.

stay tuned for the minor release patch

**Comment by fenos** [comment]
Probably this is related to #19 as the algorithm to parse the map type it’s not entirely correct, I’ll give it a go to push a proper fix soon

**Comment by fenos** [comment]
@yeenow123 This is now fixed, with the patch version 1.1.2
